AmazonChimeSDK/AmazonChimeSDK/audiovideo/AudioVideoConfiguration.swift

```diff
@@ -10,47 +10,101 @@ import Foundation
-        self.init(audioMode: .stereo48K, callKitEnabled: callKitEnabled, enableAudioRedundancy: true, videoMaxResolution: VideoResolution.videoResolutionHD)
+        self.init(audioMode: Self.defaultAudioMode,
+                  audioDeviceCapabilities: Self.defaultAudioDeviceCapabilities,
+                  callKitEnabled: callKitEnabled,
+                  enableAudioRedundancy: Self.defaultEnableAudioRedundancy,
+                  videoMaxResolution: Self.defaultVideoMaxResolution)
     }
 
     convenience public init(enableAudioRedundancy: Bool) {
-         self.init(audioMode: .stereo48K, callKitEnabled: false, enableAudioRedundancy: enableAudioRedundancy, videoMaxResolution: VideoResolution.videoResolutionHD)
+        self.init(audioMode: Self.defaultAudioMode,
+                  audioDeviceCapabilities: Self.defaultAudioDeviceCapabilities,
+                  callKitEnabled: Self.defaultCallKitEnabled,
+                  enableAudioRedundancy: enableAudioRedundancy,
+                  videoMaxResolution: Self.defaultVideoMaxResolution)
     }
 
     convenience public init(videoMaxResolution: VideoResolution) {
-        self.init(audioMode: .stereo48K, callKitEnabled: false, enableAudioRedundancy: true, videoMaxResolution: videoMaxResolution)
+        self.init(audioMode: Self.defaultAudioMode,
+                  audioDeviceCapabilities: Self.defaultAudioDeviceCapabilities,
+                  callKitEnabled: Self.defaultCallKitEnabled,
+                  enableAudioRedundancy: Self.defaultEnableAudioRedundancy,
+                  videoMaxResolution: videoMaxResolution)
     }
 
     convenience public init(audioMode: AudioMode, callKitEnabled: Bool) {
-        self.init(audioMode: audioMode, callKitEnabled: callKitEnabled, enableAudioRedundancy: true, videoMaxResolution: VideoResolution.videoResolutionHD)
+        self.init(audioMode: audioMode,
+                  audioDeviceCapabilities: Self.defaultAudioDeviceCapabilities,
+                  callKitEnabled: callKitEnabled,
+                  enableAudioRedundancy: Self.defaultEnableAudioRedundancy,
+                  videoMaxResolution: Self.defaultVideoMaxResolution)
     }
 
     convenience public init(audioMode: AudioMode, callKitEnabled: Bool, enableAudioRedundancy: Bool) {
-        self.init(audioMode: audioMode, callKitEnabled: callKitEnabled, enableAudioRedundancy: enableAudioRedundancy, videoMaxResolution: VideoResolution.videoResolutionHD)
+        self.init(audioMode: audioMode,
+                  audioDeviceCapabilities: Self.defaultAudioDeviceCapabilities,
+                  callKitEnabled: callKitEnabled,
+                  enableAudioRedundancy: enableAudioRedundancy,
+                  videoMaxResolution: Self.defaultVideoMaxResolution)
     }
 
-    public init(audioMode: AudioMode, callKitEnabled: Bool, enableAudioRedundancy: Bool, videoMaxResolution: VideoResolution) {
+    public init(audioMode: AudioMode = AudioVideoConfiguration.defaultAudioMode,
```

Comment:
Did you verify if this initializer support obj-c?

Reply:
Yeh this initializer still shows up. But given that if we add a new field in the future, this initializer would have to become a convenience initializer that delegates to the new designated initializer, which would require us to remove those default parameters in the signature. To avoid this, I'll just remove the default parameters now so that we can stick to the current pattern.